### ISSUE TITLE
Check if `aria-haspopup="dialog"` is needed

### ISSUE THREAD

**Issue by Barsnes** [digdir/designsystem#4456]
Today we write:
> Note that the button that opens the dialog should have the aria-haspopup="dialog" attribute.

We need to investigate if we actually need this.

See: https://designsystemet.no/en/components/docs/dialog/code/#html

**Comment by eirikbacker** [comment]
Quick test in VoiceOver and NVDA: this property has effect and improves the user experience, so we should keep it :)

**Comment by eirikbacker** [comment]
Fixed by #4409
